### PR TITLE
Fixed and added support for dataUri in form data

### LINKED ISSUE #25790: [Android] FormData does not support data URIs
React Native version: Expo 33 / React Native 0.59.8

## Steps To Reproduce

1. Call `formData.append` with a data URI as the `uri`
2. Make XHR request and send FormData

```js
const dataUri = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVQYV2NgYAAAAAMAAWgmWQ0AAAAASUVORK5CYII=';
const formData = new FormData();
formData.append('file', {
  name: 'myfile',
  type: 'image/png',
  uri: dataUri,
});

const request = new XMLHttpRequest();
request.open('POST', 'http://httpbin.org/post');
request.onreadystatechange = function() {
  if (request.readyState === 4) {
    if (request.status === 200) {
      console.log("SUCCESS")
      // will succeed on iOS
    } else {
      console.log(request.responseText);
      // response on Android will be "Could not retrieve file for uri data:image/png;base64,iVBO..."
    }
  }
};
request.send(formData);
```

Describe what you expected to happen:

The request would succeed (as it does on iOS) but instead (on Android only), you get this error:

>Could not retrieve file for uri data:image/png;base64,iVB...

Snack: https://snack.expo.io/@maxholderc5/cd0ef5

It appears that the [Android implementation](https://github.com/facebook/react-native/blob/cf77067f0c4eda30f814aafa45ddf962eadcfbdb/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L700-L709) assumes this will always be a file, but unless I've misunderstood, passing a data URI here should also be valid (at least, it seems to work on iOS).

A workaround for my specific scenario (in case it is helpful for others running into this) was to write a temporary file and provide its filepath instead of the data URI directly, e.g. with Expo:

```js
const tempFileDirectory = `${FileSystem.cacheDirectory}uploadTmp`
await FileSystem.makeDirectoryAsync(tempFileDirectory, { intermediates: true })
const tempFilePath = `${tempFileDirectory}${Date.now()}.png`
const base64EncodedData = dataUri.split(',')[1]
await FileSystem.writeAsStringAsync(tempFilePath, base64EncodedData, {
  encoding: FileSystem.EncodingType.Base64,
})
// ...
formData.append(name, {
  name: "myfile",
  type: "image/png",
  uri: tempFilePath,
})
// ...
await FileSystem.deleteAsync(tempFilePath)
```

### PR DESCRIPTION
**Pull request by hetanthakkar1**
## Summary

Fixes #25790. The issue resulted because the [getFileInputStream() method of RequestBodyUtil failed to return an inputStream of the given dataUri ](https://github.com/facebook/react-native/blob/16397e0d3c0dd3374ddb642599725ca41f092a8a/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L427) This happened because the openInputStream() method of [context.getContentResolver() expects a content or a file](https://developer.android.com/reference/android/content/ContentResolver#openInputStream(android.net.Uri)) in its parameter but instead received dataUri which resulted in FileNotFoundException.

issue

. Therefore it makes sense to provide a file path to the openInputStream method instead of dataUri. 

**Solution**:

1) Converted the dataUri to a bitmap  and then stored the bitmap image in a file to get the file path. [Code Reference](https://stackoverflow.com/a/37706248/10033505)
2) This file path is  passed to the openInputStream() method [Code Referrence](https://stackoverflow.com/a/16168087/10033505)
3) Once the FileInputStream is generated, I deleted the temporarily created file.

This issue has been unresolved for quite some time now, so resolving this PR would be greatly appreciated.
## Changelog

[Android] [Added] - Support for dataUri in form data

### CODE REVIEW

**Comment by hetanthakkar1** [comment]
@kelset @cortinico @lunaleaps can you please review this pr?

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java [hunk 1]
```diff
@@ -52,17 +56,32 @@ public static boolean isGzipEncoding(@Nullable final String encodingType) {
...
       if (fileContentUri.getScheme().startsWith("http")) {
         return getDownloadFileInputStream(context, fileContentUri);
       }
+
+      if (fileContentUriStr.startsWith("data:")) {
+        byte[] decodedDataUrString = Base64.decode(fileContentUriStr.split(",")[1], Base64.DEFAULT);
+        Bitmap bitMap = BitmapFactory.decodeByteArray(decodedDataUrString, 0, decodedDataUrString.length);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bitMap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);  
+        String path = MediaStore.Images.Media.insertImage(context.getContentResolver(), bitMap, "Title", null);
```
Comment (ryancat) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:72]: I tried to look up for more options, and datauri seems not supported out of the box. I agree to create a temp file and use the input stream like so is the correct solution. However, I suggest:

1. Evaluate any impact on large image files
2. Checkout the next method in this file, [getDownloadFileInputStream](https://github.com/facebook/react-native/blob/3199c5801ca41476e4756dbcd00a154dd87b50a8/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java#L91), where it also created a temp file and return the inputstream. Can we somehow re-use the implementationt here?

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java]: @ryancat  Thanks for the review! I've now converted the dataUri in bitmap and then attached an inputStream to the compressed bitmap. This way we won't have to store the image temporarily. I think converting the dataUri to Bitmap is necessary as there is no method that lets us convert the dataUri to inputStream directly. So regarding large size images, converting them to bitmap is the only(efficient) step we are left with, and the conversion is handled by in-built java function

**Comment by hetanthakkar1** [comment]
@ryancat The test which is failing, is related to my code changes?

**Comment by hetanthakkar1** [comment]
@cortinico Can you please review!?

File: ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java [hunk 2]
```diff
@@ -58,6 +62,17 @@ public static boolean isGzipEncoding(@Nullable final String encodingType) {
       if (fileContentUri.getScheme().startsWith("http")) {
         return getDownloadFileInputStream(context, fileContentUri);
       }
+
+      if (fileContentUriStr.startsWith("data:")) {
+        byte[] decodedDataUrString = Base64.decode(fileContentUriStr.split(",")[1], Base64.DEFAULT);
+        Bitmap bitMap = BitmapFactory.decodeByteArray(decodedDataUrString, 0, decodedDataUrString.length);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bitMap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);  
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
```
Comment (cortinico) [inline ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:71]: Is this unused?

Comment (hetanthakkar1) [reply ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java]: @cortinico yes that variable is unused and i forgot to remove that before commiting.

**Pull request closed without merging**